openhands-sdk/openhands/sdk/agent/agent.py

```diff
@@ -104,15 +98,24 @@ def init_state(
             event = SystemPromptEvent(
                 source="agent",
                 system_prompt=TextContent(text=self.system_message),
+                # Always include security_risk field in tools
                 tools=[
-                    t.to_openai_tool(
-                        add_security_risk_prediction=self._add_security_risk_prediction
-                    )
+                    t.to_openai_tool(add_security_risk_prediction=True)
                     for t in self.tools_map.values()
                 ],
             )
             on_event(event)
 
+        security_analyzer_event = SecurityAnalyzerConfigurationEvent.from_analyzer(
+            analyzer=self.security_analyzer
+        )
+
+        if (
+            len(security_analyzer_configuration_events) == 0
+            or security_analyzer_event != security_analyzer_configuration_events[-1]
+        ):
+            on_event(security_analyzer_event)
```

Comment:
It seems to me, this security analyzer event solely act as some sort of flag?

Can we add a new field to ConversationState to keep this information instead of making it an event?

Reply:
So with this PR we always store `security_risk` param regardless of the security analyzer. So without knowing which analyzer was active at the time of the action being generated, a stored security_risk value is ambiguous: you can’t tell if it was enforced, advisory, or ignored.


SecurityAnalyzerConfigurationEvent provides a timestamped, minimal audit trail of the analyzer type in effect when the action was proposed. This is especially important as we don't record user confirmation events, they are implicitly assumed if the action was executed

Reply:
How about we keep a variable inside State that record the latest timestamp and the security analyzer type instead of emitting an event?

I'm hesitant about throwing this into events as this make the logic here very hard to parse

Reply:
So one reason I can think of adding the information to agent event history, instead of the conversation state, is that the `security_risk` is a default specific to the `Agent` class. Other folks who build their own agent implementations will use the same conversation state but may not have a `security_risk` requirement the way we do. So for specificity is makes sense to isolate that customization to the history for the default agent, rather than the shared state object

Reply:
yep! i think that's a more reasonable thing to do (although this means we are making breaking change 1 day after cutting the v1 release haha 🤣 )